tools/windows-node-installer/pkg/cloudprovider/azure/vm.go

```diff
@@ -464,6 +470,13 @@ func (az *AzureProvider) constructStorageProfile(imageId string) (storageProfile
 	return storageProfile
 }
 
+// getWorkerBackendPoolID gets the backend pool id of the worker node loadbalancer
+func (az *AzureProvider) getWorkerBackendPoolID() *string {
+	var wbpi = fmt.Sprintf("/subscriptions/%s/resourceGroups/%[2]s-rg/providers/"+
+		"Microsoft.Network/loadBalancers/%[2]s/backendAddressPools/%[2]s", az.subscriptionID, az.infraID)
```

Comment:
Do you have checks for `az.subscriptionID` and `az.infraID`, can they be nil? What would happen to Loadbalancer field when it has nil value?

Reply:
in that case, we need to add a global check for infraID and subscriptionID. I assumed that they're being checked before assignment. Should i add a global check for infraID and subscriptionID ?